src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1,1161 +1,608 @@
+if (!post) {
+  Astro.redirect("/404")
+}
 ---
-<h1>{title}</h1>
-<ul>
-    {orders.map(order => {
-        return <li>{/* do something with each order */}</li>
-    })}
-</ul>
+<html>
+  <h1>{post.name}</h1>
+</html>
 ```
 
-### `Astro.preferredLocale`
-
-<p>
-
-**Type:** `string | undefined`<br />
-<Since v="3.5.0" />
-</p>
-
-`Astro.preferredLocale` is a computed value that represents the preferred locale of the user. 
-
-It is computed by checking the configured locales in your `i18n.locales` array and locales supported by the users's browser via the header `Accept-Language`. This value is `undefined` if no such match exists.
-
-This property is only available when building for SSR (server-side rendering) and should not be used for static sites.
-
-### `Astro.preferredLocaleList`
-
-<p>
-
-**Type:** `string[] | undefined`<br />
-<Since v="3.5.0" />
-</p>
-
-`Astro.preferredLocaleList` represents the array of all locales that are both requested by the browser and supported by your website. This produces a list of all compatible languages between your site and your visitor. 
-
-If none of the browser's requested languages are found in your locales array, then the value is `[]`: you do not support any of your visitor's preferred locales.
-
-If the browser does not specify any preferred languages, then this value will be [`i18n.locales`](/en/reference/configuration-reference/#i18nlocales): all of your supported locales will be considered equally preferred by a visitor with no preferences. 
-
-This property is only available when building for SSR (server-side rendering) and should not be used for static sites.
+See also: [`params`](/en/reference/routing-reference/#params)
 
-### `Astro.currentLocale`
+### `url`
 
 <p>
 
-**Type:** `string | undefined`<br />
-<Since v="3.5.6" />
+**Type:** `URL`<br />
+<Since v="1.0.0" />
 </p>
 
-The locale computed from the current URL, using the syntax specified in your `locales` configuration. If the URL does not contain a `/[locale]/` prefix, then the value will default to `i18n.defaultLocale`.
-
-### `Astro.getActionResult()`
+`url` is a [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) object constructed from the current `request.url` URL string value. It is useful for interacting with individual properties of the request URL, like pathname and origin. 
 
-<p>
-**Type:** `(action: TAction) => ActionReturnType<TAction> | undefined`<br />
-<Since v="4.15.0" />
-</p>
+`Astro.url` is equivalent to doing `new URL(Astro.request.url)`.
 
-`Astro.getActionResult()` is a function that returns the result of an [Action](/en/guides/actions/) submission. This accepts an action function as an argument (e.g. `actions.logout`) and returns a `data` or `error` object when a submission is received. Otherwise, it will return `undefined`.
+`url` will be `localhost` in dev mode for all on-demand rendered routes, and for prerendered routes when [`site`](/en/reference/configuration-reference/#site) is not configured.
```

Comment:
I think I understand this right. The second segment is also in relation to dev mode only? I found it a little confusing during the first read. Probably just me, but thought it was worth a mention

Reply:
I actually was concerned the original was confusing. 😅  Currently it says:

> Astro.url will be localhost in dev mode if [site](https://docs.astro.build/en/reference/configuration-reference/#site) is not configured for static sites, and for on-demand rendered sites using server or hybrid output.

I am assuming it's dev mode for all? I wouldn't be localhost in production, right?